server/api/cli/cli.controller.js

```diff
@@ -39,17 +39,22 @@ exports.run = function (req, res) {
   dirName += dateTime.getFullYear()
   const execFile = req.body.fileName.replace('.c', '')
   const cmd = `"${dirName}/${execFile}"`
-  exec(cmd, { timeout: 10000 }, // Process will time out if running for > 10 seconds.
+  var cp = exec(cmd, { timeout: 10000 }, // Process will time out if running for > 10 seconds.
     (error, stdout, stderr) => {
       if (error) {
-        return res.send(200, error)
+        return res.send(200, {"info":error, "process":cp.pid});
```

Comment:
Any idea why this might be? The error that is caught when an infinite loop is run seemed to just be a json object with the file name..wasn't able to tell why it was coming back with an error before the timeout